File: MoniGuardAPI/Guest.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace MoniGuardAPI;

/// <summary>
/// 来宾信息。
/// </summary>
/// <param name="guestId">来宾 ID。</param>
/// <param name="sceneId">来宾所属场景 ID。</param>
/// <param name="name">来宾名称。</param>
/// <param name="createdAt">来宾创建时间。</param>
/// <param name="isWhitelisted">来宾是否已经加入白名单。</param>
/// <param name="faceEncoding">来宾的人脸数据。</param>
public class Guest(int guestId, int sceneId, string name, DateTime createdAt, bool isWhitelisted, byte[]? faceEncoding)
{
    /// <summary>
    /// 创建来宾信息的构造方法。
    /// </summary>
    /// <inheritdoc cref="Guest"/>
    public Guest(int sceneId, string name, bool isWhitelisted) :this(0, sceneId, name, DateTime.Now, isWhitelisted, null) { }

    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int GuestId { get; set; } = guestId;

    /// <summary>
    /// 来宾所属场景 ID。
    /// </summary>
    [ForeignKey("Scene")]
    public int SceneId { get; set; } = sceneId;

    /// <summary>
    /// 来宾名称。
    /// </summary>
    [Length(2, 50)]
    public string Name { get; set; } = name;

    /// <summary>
    /// 来宾创建时间。
    /// </summary>
    public DateTime CreatedAt { get; set; } = createdAt;

    /// <summary>
    /// 来宾是否已经加入白名单。
    /// </summary>
    public bool IsWhitelisted { get; set; } = isWhitelisted;

    /// <summary>
    /// 来宾的人脸数据。
    /// </summary>
    public byte[]? FaceEncoding { get; set; } = faceEncoding;

}

/// <summary>
/// 来宾数据传输对象。
/// </summary>
/// <param name="name">来宾名称。</param>
/// <param name="isWhitelisted">是否加入白名单。该值可空，空值表示不加入。</param>
[method:JsonConstructor]
public struct GuestDto(string name, bool? isWhitelisted)
{
    /// <summary>
    /// 来宾名称。
    /// </summary>
    public string Name { get; } = name;

    /// <summary>
    /// 是否加入白名单。该值可空，空值表示不加入。
    /// </summary>
    public bool? IsWhitelisted { get; } = isWhitelisted;
}

File: moni_guard_api/guest.py
from datetime import datetime


class Guest:
    def __init__(self, guest_id: int, scene_id: int, name: str, created_at: datetime, is_whitelisted: bool):
        """
        :param guest_id: 访客 ID。
        :param scene_id: 场景 ID。
        :param name: 访客名称。
        :param created_at: 访客创建时间。
        :param is_whitelisted: 访客是否允许。
        """
        self.guest_id = guest_id
        self.scene_id = scene_id
        self.name = name
        self.created_at = created_at
        self.is_whitelisted = is_whitelisted

    def __str__(self):
        return f"Guest(guest_id={self.guest_id}, scene_id={self.scene_id}, name={self.name}, created_at={self.created_at}, is_whitelisted={self.is_whitelisted})"

    def __repr__(self):
        return str(self)

    @staticmethod
    def from_json(json):
        return [Guest(guest_id=guest_object['guestId'], scene_id=guest_object['sceneId'], name=guest_object['name'],
                      created_at=datetime.strptime(guest_object['createdAt'][:-1], "%Y-%m-%dT%H:%M:%S.%f"),
                      is_whitelisted=guest_object['isWhitelisted']) for guest_object in json]
